session_test.go

```diff
@@ -158,3 +159,23 @@ func TestSession_sendPings(t *testing.T) {
 		}
 	}
 }
+
+// This test is to ensure that there is no deadlock if Close()
+// is called while startPings goroutine is running.
+func TestSession_CloseDeadlock(t *testing.T) {
```

Comment:
I'm not a fan of tests that would block indefinitely as a way of reporting there is an error.
If you can find a way to abort it after a reasonable time and call `t.Fatal` with a meaningful message that would be nice.

Reply:
Done.